### PR TITLE
fix: allow test flag for update

### LINKED ISSUE #6434: Allow configurable update
Allow a configurable update mechanism for upgrade-8 so that we can use test using test funds that are separate from the mainnet funds

### PR DESCRIPTION
**Pull request by arirubinstein**
closes: #6434
refs: #XXXX

## Description

Adds a new upgrade handler for a test config

### CODE REVIEW

File: golang/cosmos/app/app.go [hunk 1]
```diff
@@ -788,6 +767,41 @@ func NewAgoricApp(
...
+			return vm, err
+		}
+
+		// Set bean count
+		currentSwingsetParams := app.SwingSetKeeper.GetParams(ctx)
+		ctx.Logger().Info("pre-swingset upgrade", "subspace", swingset.ModuleName, "params", currentSwingsetParams)
+
+		for i := 0; i < len(currentSwingsetParams.BeansPerUnit); i++ {
+			if currentSwingsetParams.BeansPerUnit[i].Key == swingsettypes.BeansPerBlockComputeLimit {
+				currentSwingsetParams.BeansPerUnit[i].Beans = sdk.NewUint(6500000000)
```
Comment (JimLarson) [inline golang/cosmos/app/app.go]: Add underscores for legibility:

```
sdk.NewUint(6_500_000_000)
```

File: golang/cosmos/app/app.go [hunk 2]
```diff
@@ -788,6 +767,41 @@ func NewAgoricApp(
...
+
+		// Set bootstrap
+		switch upgradeName {
+		case upgradeName:
+			currentSwingsetParams.BootstrapVatConfig = "@agoric/vats/decentral-main-psm-config.json"
+		case upgradeNameTest:
+			currentSwingsetParams.BootstrapVatConfig = "@agoric/vats/decentral-test-psm-config.json"
+		default:
+			return vm, fmt.Errorf("invalid upgrade name")
+		}
```
Comment (JimLarson) [inline golang/cosmos/app/app.go]: Consider switching to compute the config file name and consolidating thereafter:

```
var configPath string
switch name {
case upgradeName:
        configPath = "@agoric/vats/decentral-main-psm-config.json"
case upgradeNameTest:
        configPath = "@agoric/vats/decentral-test-psm-config.json"
default:
        return vm, fmt.Errorf("invalid upgrade name: %s", name)
}
currentSwingsetParams.BootstrapVatConfig = configPath
```

File: golang/cosmos/app/app.go [hunk 3]
```diff
@@ -788,6 +767,41 @@ func NewAgoricApp(
 	return app
 }
 
+func upgrade8Handler(app *GaiaApp, upgradeName string) func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
```
Comment (JimLarson) [inline golang/cosmos/app/app.go]: The `upgradeName` here shadows the constant of the same name - rename to `name` here, or rename the constant to `upgradeMainName` or something.

File: packages/vats/decentral-test-psm-config.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        "anchorAssets": [
+          {
+            "denom": "ibc/usdc1234"
+          }
+        ],
+        "economicCommitteeAddresses": {
+          "gov1": "agoric1ldmtatp24qlllgxmrsjzcpe20fvlkp448zcuce",
+          "gov2": "agoric140dmkrz2e42ergjj7gyvejhzmjzurvqeq82ang",
+          "gov3": "agoric1w8wktaur4zf8qmmtn3n7x3r0jhsjkjntcm3u6h"
+        }
```
Comment (JimLarson) [inline packages/vats/decentral-test-psm-config.json:18]: Confirm that this is the only intended different from the main config.

**Review by JimLarson — CHANGES_REQUESTED** [review]

File: golang/cosmos/app/app.go (hunk 2 above)
Comment (JimLarson) [reply golang/cosmos/app/app.go]: This is a very weak suggestion - it doesn't declutter as much as I had hoped.

File: packages/vats/decentral-test-psm-config.json (hunk 4 above)
Comment (arirubinstein) [reply packages/vats/decentral-test-psm-config.json:18]: yep, and ultimately we'll patch up the denom in anchorAssets to axelar test once that's wired up

**Review by JimLarson — APPROVED** [review]

**Review by dckc — APPROVED** [review]
looks good.

I'll note:
refs: #6284

**Pull request merged**